contracts/core/interfaces/ICore.sol

```diff
@@ -0,0 +1,175 @@
+/*
+    Copyright 2018 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.4.24;
+
+/**
+ * @title ICore
+ * @author Set Protocol
+ *
+ * The ICore Contract defines all the functions exposed in the Core through its
+ * various extensions and is a light weight way to interact with the contract.
+ */
+
+ interface ICore {
+
+    /**
+     * Set vaultAddress. Can only be set by owner of Core.
+     *
+     * @param  _vaultAddress   The address of the Vault
+     */
+    function setVaultAddress(
+        address _vaultAddress
+    )
+        external;
+
+    /**
+     * Set transferProxyAddress. Can only be set by owner of Core.
+     *
+     * @param  _transferProxyAddress   The address of the TransferProxy
+     */
+    function setTransferProxyAddress(
+        address _transferProxyAddress
+    )
+        external;
+
+    /**
+     * Add a factory to the mapping of tracked factories.
+     *
+     * @param  _factoryAddress   The address of the SetTokenFactory to enable
+     */
+    function enableFactory(
+        address _factoryAddress
+    )
+        external;
+
+    /**
+     * Disable a factory in the mapping of tracked factories.
+     *
+     * @param  _factoryAddress   The address of the SetTokenFactory to disable
+     */
+    function disableFactory(
+        address _factoryAddress
+    )
+        external;
+
+    /**
+     * Disable a set token in the mapping of tracked set tokens.
+     *
+     * @param  _setAddress   The address of the SetToken to remove
+     */
+    function disableSet(
+        address _setAddress
+    )
+        external;
+
+    /**
+     * Issue
+     *
+     * @param  _setAddress   Address of set to issue
+     * @param  _quantity     Quantity of set to issue
```

Comment:
We should add that the quantity to issue should be a multiple of the naturalUnit of the Set